docs/griptape-framework/structures/task-memory.md

```diff
@@ -1,89 +1,320 @@
+
+## Task Memory Client
+
+The [TaskMemoryClient](../griptape-tools/official-tools/task-memory-client.md) is a Tool that allows an Agent to interact with Task Memory. It has the following methods:
+
+- `query`: Retrieve the content of an Artifact stored in Task Memory.
+- `summarize`: Summarize the content of an Artifact stored in Task Memory.
+
+Let's add `TaskMemoryClient` to the Agent and run the same task:
+
+```python
+from griptape.structures import Agent
+from griptape.tools import Calculator, TaskMemoryClient
+
+# Create an agent with the Calculator tool
+agent = Agent(tools=[Calculator(off_prompt=True), TaskMemoryClient(off_prompt=False)])
+
+agent.run("What is the square root of 12345?")
+```
 
-By default, Griptape augments all tool outputs with [TaskMemory](../../reference/griptape/memory/task/task_memory.md) but you can override at the structure, task, or tool activity level.
+```
+[04/26/24 13:13:01] INFO     ToolkitTask 5b46f9ef677c4b31906b48aba3f45e2c
+                             Input: What is the square root of 12345?
+[04/26/24 13:13:07] INFO     Subtask 611d98ea5576430fbc63259420577ab2
+                             Thought: To find the square root of 12345, I can use the Calculator action with the expression "12345 ** 0.5".
+                             Actions: [{"name": "Calculator", "path": "calculate", "input": {"values": {"expression": "12345 ** 0.5"}}, "tag": "sqrt_12345"}]
+[04/26/24 13:13:08] INFO     Subtask 611d98ea5576430fbc63259420577ab2
+                             Response: Output of "Calculator.calculate" was stored in memory with memory_name "TaskMemory" and artifact_namespace
+                             "7554b69e1d414a469b8882e2266dcea1"
+[04/26/24 13:13:15] INFO     Subtask 32b9163a15644212be60b8fba07bd23b
+                             Thought: The square root of 12345 has been calculated and stored in memory. I can retrieve this value using the TaskMemoryClient action with
+                             the query path, providing the memory_name and artifact_namespace as input.
+                             Actions: [{"tag": "retrieve_sqrt", "name": "TaskMemoryClient", "path": "query", "input": {"values": {"memory_name": "TaskMemory",
+                             "artifact_namespace": "7554b69e1d414a469b8882e2266dcea1", "query": "What is the result of the calculation?"}}}]
+[04/26/24 13:13:16] INFO     Subtask 32b9163a15644212be60b8fba07bd23b
+                             Response: The result of the calculation is 111.1080555135405.
+[04/26/24 13:13:17] INFO     ToolkitTask 5b46f9ef677c4b31906b48aba3f45e2c
+                             Output: The square root of 12345 is approximately 111.108.
+```
+
+Note that on the `TaskMemoryClient` we've set `off_prompt` to `False` so that the results of the query can be returned directly to the LLM. If we had kept it as `True`, the results would have been stored back Task Memory which would've put us back to square one.
```

Comment:
nit: I'd put this comment up above the example as it was my first question when reading the example and was then confused until I got here.